[Bug] Crash : Ankidroid crashes on checking media.

###### Reproduction Steps
- Updated from 2.15.6 (Play store ver.) to 2.16alpha58
 and when i opened a card i started getting "failed to load : failed to load ...."
![Screenshot_2022-04-16-17-34-50-744_com.ichi2.anki.png](https://user-images.githubusercontent.com/63903377/163674249-884f95df-96c1-44f1-8a0e-8a5fdc0edefd.png)
- So i decided to use check media which instantly crashed the app.
- After that i decided to force sync from anki web to ankidroid which also crashed the app.
- Decided to check if my library was corrupted for some reason or not by downloading a parallel of 2.15.6 on which it seemed to work perfectly, updated the parallel to alpha58 and it was also crashing.
- The alpha also doesn't seem to be detecting the backups neither is it creating them properly from what i can tell.
![Screenshot_2022-04-16-17-29-38-032_ru.zdevs.zarchiver.pro.png](https://user-images.githubusercontent.com/63903377/163674472-24c1d8ce-3c83-4239-b7cb-ed7f6b001482.png)
The only one which was good on 16/04 was made by the stable parallel.


- Just to confirm if the problem was from this alpha or from before this I decided to test on alpha 56 parallel and got the same issues uninstalled the parallel and installed alpha 55 parallel and there wasn't a crash this time, decided to update 55 to 56 and test again and there wasn't a crash, same with from 56 to 58.
- ~~Btw I did try to reinstall main alpha58 directly when i was originally getting the crash before the above point happened and the crashes still happened.~~ I thought i did but seems like i hadn't installed it directly and when I just did it seems to work fine on installing directly and this only happens if you update directly from 2.15.6 to 2.16 Alpha56/58/59.
- Have confirmed that it doesn't crash anymore on main after uninstalling 58 and installing 55 then 56 and 58 again after that.
- ~~Backup still don't seem to be being created properly but am not sure yet, so will confirm after I am.~~ It is detecting backups again and the backups are being created properly.

- Also half of this happened while writing this so it might seem like a story.

###### Expected Result

App checks media , directly syncs from anki web to ankidroid , Backup being created properly.


###### Actual Result

App crashes if you have updated directly to alpha 58 or installing alpha58 directly over current library.

###### Debug info
AnkiDroid Version = 2.16alpha58

Android Version = 11

Manufacturer = Xiaomi

Model = Redmi Note 9 Pro Max

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 11; Redmi Note 9 Pro Max Build/RKQ1.200826.002; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/100.0.4896.88 Mobile Safari/537.36

ACRA UUID = d9199133-a62c-4572-ad4f-899c4d0b5307

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true


###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

Crash during check media:

https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/d41c144d-1c3f-476d-8193-761f1e1375c4

```
java.lang.RuntimeException: An error occurred while executing doInBackground()
at android.os.AsyncTask$4.done(AsyncTask.java:415)
at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
at java.util.concurrent.FutureTask.run(FutureTask.java:271)
at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:305)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
at java.lang.Thread.run(Thread.java:923)
Caused by: java.lang.NullPointerException: Attempt to get length of null array
at com.ichi2.libanki.Media._changes(Media.java:11)
at com.ichi2.libanki.Media._logChanges(Media.java:1)
at com.ichi2.libanki.Media.findChanges(Media.java:3)
at com.ichi2.async.CollectionTask$CheckMedia.task(CollectionTask.java:4)
at com.ichi2.async.CollectionTask$CheckMedia.task(CollectionTask.java:1)
at com.ichi2.async.CollectionTask.actualDoInBackground(CollectionTask.java:14)
at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:2)
at com.ichi2.async.CollectionTask.doInBackground(CollectionTask.java:1)
at android.os.AsyncTask$3.call(AsyncTask.java:394)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
... 4 more
```

--- a/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt
+++ b/AnkiDroid/src/main/java/com/ichi2/anki/DeckPicker.kt
@@ -471,13 +471,14 @@ open class DeckPicker : NavigationDrawerActivity(), StudyOptionsListener, SyncEr
     @KotlinCleanup("remove parameters named _")
     @KotlinCleanup("return early and remove else")
     fun requestStoragePermission() {
+        val storagePermissions = arrayOf(
+            Manifest.permission.WRITE_EXTERNAL_STORAGE,
+            Manifest.permission.READ_EXTERNAL_STORAGE
+        )
         if (mClosedWelcomeMessage) {
             // DEFECT #5847: This fails if the activity is killed.
             // Even if the dialog is showing, we want to show it again.
-            ActivityCompat.requestPermissions(
-                this, arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE),
-                REQUEST_STORAGE_PERMISSION
-            )
+            ActivityCompat.requestPermissions(this, storagePermissions, REQUEST_STORAGE_PERMISSION)
         } else {
             Timber.i("Displaying initial permission request dialog")
             // Request storage permission if we don't have it (e.g. on Android 6.0+)
@@ -488,10 +489,7 @@ open class DeckPicker : NavigationDrawerActivity(), StudyOptionsListener, SyncEr
                 .positiveText(R.string.dialog_ok)
                 .onPositive { _: MaterialDialog?, _: DialogAction? ->
                     mClosedWelcomeMessage = true
-                    ActivityCompat.requestPermissions(
-                        this, arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE),
-                        REQUEST_STORAGE_PERMISSION
-                    )
+                    ActivityCompat.requestPermissions(this, storagePermissions, REQUEST_STORAGE_PERMISSION)
                 }
                 .cancelable(false)
                 .canceledOnTouchOutside(false)
@@ -712,8 +710,8 @@ open class DeckPicker : NavigationDrawerActivity(), StudyOptionsListener, SyncEr
 
     override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
         super.onRequestPermissionsResult(requestCode, permissions, grantResults)
-        if (requestCode == REQUEST_STORAGE_PERMISSION && permissions.size == 1) {
-            if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+        if (requestCode == REQUEST_STORAGE_PERMISSION && permissions.isNotEmpty()) {
+            if (grantResults.all { it == PackageManager.PERMISSION_GRANTED }) {
                 invalidateOptionsMenu()
                 handleStartup()
             } else {

